ruby-challenges.rb

```diff
@@ -11,6 +11,26 @@
 # Expected output: ['tea', 'water', 'soda water']
 
 # Pseudo code:
+# define function extract with 2 params as array and str
+# declare a result array as []
+# loop over array with each
+  # if letter is included in array's value
+    # shovel array's value in to result array
+# return result array
+
+def extract arr, str
+  result = []
+  arr.each do |value|
+    if value.include?(str)
+      result << value
+    end
+  end
+  result
+end
+
+p extract(beverages_array, filter_letter_o)
+p extract(beverages_array, filter_letter_t)
+
```

Comment:
Nice approach! When refactoring think about ruby methods that return arrays or subsets of arrays.